assessment/cog-memory-leaks.robot

```diff
@@ -0,0 +1,104 @@
+    ${environment}=    Catenate
+    ...    COG_PLATFORM_WL_VIEW_HEIGHT=720 COG_PLATFORM_WL_VIEW_WIDTH=1280
+    ...    COG_PLATFORM_WL_VIEW_MAXIMIZE=0 WAYLAND_DISPLAY=wayland-1
+    ...    XDG_RUNTIME_DIR=/run/user/0
+    ${url}=    Set Variable    https://pixijs.com/8.x/examples/mesh-and-shaders/instanced-geometry
+    Execute Command In Terminal    export ${environment}
+    Execute Command In Terminal    mem_test "cog ${url}" ${time} 50    ${cmd_timeout}s
+    Set Prompt For Terminal    root@${HOSTNAME}:${WORKDIR}#
+    Execute Command In Terminal    cd ..
+
+COG001.002 Check memory usage on light application
+    [Documentation]    This test measures memory usage while running a
+    ...    memory-light web application in Cog browser.
+    ${variables}=    Get Variables
+    IF    "\${TIME}" not in $variables
+        ${time}=    Set Variable    60
+    END
+    ${cmd_timeout}=    Evaluate    int($TIME) + 500
+    Variable Should Exist    ${HOSTNAME}
+    Variable Should Exist    ${WORKDIR}
+    Set Prompt For Terminal    root@${HOSTNAME}:${WORKDIR}/memory_light#
+    Execute Command In Terminal    mkdir memory_light && cd memory_light
+    ${environment}=    Catenate
+    ...    COG_PLATFORM_WL_VIEW_HEIGHT=720 COG_PLATFORM_WL_VIEW_WIDTH=1280
+    ...    COG_PLATFORM_WL_VIEW_MAXIMIZE=0 WAYLAND_DISPLAY=wayland-1
+    ...    XDG_RUNTIME_DIR=/run/user/0
+    ${url}=    Set Variable    https://www.timeanddate.com/worldclock/
+    Execute Command In Terminal    export ${environment}
+    Execute Command In Terminal    mem_test "cog ${url}" ${time} 50    ${cmd_timeout}s
+    Set Prompt For Terminal    root@${HOSTNAME}:${WORKDIR}#
+    Execute Command In Terminal    cd ..
+
+COG002.001 Check for memory leaks using Heaptrack
+    [Documentation]    This test uses Heaptrack to check for memory leaks in Cog
+    ...    browser.
+    ${variables}=    Get Variables
+    IF    "\${TIME}" not in $variables
+        ${time}=    Set Variable    60
+    END
+    ${cmd_timeout}=    Evaluate    int($TIME) + 30
+    Variable Should Exist    ${HOSTNAME}
+    Variable Should Exist    ${WORKDIR}
+    Set Prompt For Terminal    root@${HOSTNAME}:${WORKDIR}/memleaks#
+    Execute Command In Terminal    mkdir memleaks && cd memleaks
+    ${environment}=    Catenate
+    ...    COG_PLATFORM_WL_VIEW_HEIGHT=720 COG_PLATFORM_WL_VIEW_WIDTH=1280
+    ...    COG_PLATFORM_WL_VIEW_MAXIMIZE=0 WAYLAND_DISPLAY=wayland-1
+    ...    XDG_RUNTIME_DIR=/run/user/0
+    ${url}=    Set Variable    https://pixijs.com/8.x/examples/mesh-and-shaders/instanced-geometry
+    Execute Command In Terminal    export ${environment}
+    #    Heaptrack cannot run in the background
+    Write Into Terminal    heap_analyze "cog ${url}"
+    Press Enter
+    Sleep    ${time}
+    Press Key N Times    1    ${CTRL_C}
+    Set Prompt For Terminal    root@${HOSTNAME}:${WORKDIR}#
+    Execute Command In Terminal    cd ..
+
+
+*** Keywords ***
+Clean Up
```

Comment:
@PLangowski keyword is not used anywhere. If it's not required, delete it. If the opposite, we should set a prompt based on the platform config. Right now, it would work only for Radxa CM3

Reply:
https://github.com/Dasharo/open-source-firmware-validation/pull/421/commits/242cff74a06b9f6db2c258471f98701ce89af088#diff-b2cf75a1510fd482b801bbc19af389b679d1f98b38e0eb2281482be3e7e7ef5dL101